scim/client.bal

```diff
@@ -0,0 +1,262 @@
+import ballerina/oauth2;
+import ballerina/http;
+import ballerina/io;
+
+# This client is used to connect to the SCIM2 APIs of Asgardeo for Inbound User Provisioning.
+@display {label: "SCIM Client Connector"}
+public class Client {
+    private final string orgName;
+    private final string clientId;
+    private final string clientSecret;
+    private final string[] scope;
+    final oauth2:ClientOAuth2Provider provider;
+    final http:Client clientEndpoint;
+    final map<string> headers;
+
+    # Initializes the SCIM client.
+    #
+    # + orgName - The name of the organization
+    # + clientId - The client ID of the application
+    # + clientSecret - The client secret of the application
+    # + scope - permitted scopes
+    public isolated function init(string orgName, string clientId, string clientSecret, string[] scope) returns error? {
+        self.orgName = orgName;
+        self.clientId = clientId;
+        self.clientSecret = clientSecret;
+        self.scope = scope;
+
+        self.provider = new ({
+            tokenUrl: HOST_URL + TENANT_PATH + self.orgName + "/oauth2/token",
+            clientId: clientId,
+            clientSecret: clientSecret,
+            scopes: scope
+        });
+
+        self.clientEndpoint = check new (HOST_URL + TENANT_PATH + self.orgName + SCIM, {httpVersion: "1.1"});
+
+        string token = check self.provider.generateToken();
+        io:print(token);
```

Comment:
If this print statement is meant for testing purposes, shall we consider removing it?